receiver/scraperhelper/scrapercontroller.go

```diff
@@ -177,7 +162,9 @@ func (sc *controller) startScraping() {
 		for {
 			select {
 			case <-sc.tickerCh:
-				sc.scrapeMetricsAndReport(context.Background())
+				ctx, done := context.WithTimeout(context.Background(), sc.collectionInterval)
```

Comment:
I need to add more tests around this, I realised I only validate the collection interval is a positive value.